utils/http.go

```diff
@@ -135,13 +135,11 @@ func newRRFromResponse(method string, requestTrace string, r *http.Response) (*R
 	}
 	rr.Response = string(response)
 
-	if isText {
-		bodyBytes, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			return &rr, err
-		}
-		rr.Body = bodyBytes
+	bodyBytes, err := ioutil.ReadAll(r.Body)
```

Comment:
Can we revert this in favor of the described approach on https://github.com/nyaruka/courier/pull/179#discussion_r218861360?

I think we can keep this place expecting text responses and only allow other content types in specific places like the upload to WA 

@nicpottier thoughts?

Reply:
What's your worry here @norkans7? Seems having this be generic is probably just fine since callers are still going to try to parse the body one way or the other no?

Reply:
(that said, doesn't seem we need the `isText` variable anymore.

Reply:
This function being used with the code facing the aggregators for text I think it is more safe with this check

Reply:
In what way? It wasn't like we were returning an error previously when we got binary data, we just had an empty body, which seems just as likely to blow up as a binary body. All the handlers that are actually inspecting the body are parsing it in a way that should be just fine from that respect. (in that they will fail in the same way with an empty body as a binary body, such as parsing json for example) Do you have any specific example that show what you are worried about?

Reply:
I think we could be wasting time to download a 'big file' while we will not use it
really I though that was the reason to have it around

Reply:
It is, but that's more of a vestige from goflow which this originally came from I think. We are hitting known endpoints with courier and we shouldn't be getting gigungous replies so we should be fine in that respect. I think not having to repeat all our junk around headers, reading body etc, is worth it from a reuse perspective, so I think reading the body here even in the binary case is just fine.